src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -1009,21 +1011,15 @@ public static boolean isGZIPInputStream(final BufferedInputStream bufferedinput)
     
     /**
      * If `in` is a GZipped stream, wrap it into a {@link GZIPInputStream}.
-     * @param in the input stream. If it's a BufferedInputStream, the buffer must be large enough to contains the gzip signature
+     * @param in the input stream
      * @return a decompressed input stream
      * @throws IOException
      */
-    public static InputStream mayBeGZippedInputStream(final InputStream in) throws IOException {
-        if(in == null) throw new IllegalArgumentException("mayBeGZippedInputStream(null)");
+    public static InputStream gunZipIfNeeded(final InputStream in) throws IOException {
         /* BufferedInputStream is needed to decode the first bytes of the stream */
-        final BufferedInputStream bufferedinput;
-        /* it's already a buffered input stream, no need to wrap it again */
-        if(in instanceof BufferedInputStream) {
-            bufferedinput = BufferedInputStream.class.cast(in);
-        } else {
-            bufferedinput = new BufferedInputStream(in, 4);
-        }
-        if(isGZIPInputStream(bufferedinput)) {
+        final BufferedInputStream bufferedinput = new BufferedInputStream(in, 4);
```

Comment:
I'm annoyed that we will have this extra layer around all our streamed inputs....

how about making a protected wrapper class around BufferedInputStream that is only used in `VCFIteratorBuilder` and then you can have a different signature function that doesn't wrap the Stream with another buffered Stream. Since this will be the main use-case, we will generally not have the extra layer.